src/main/rx/cc2500_frsky_shared.c

```diff
@@ -72,31 +75,34 @@ static setRcDataFn *setRcData;
 
 static void initialise() {
     cc2500Reset();
-    cc2500WriteReg(CC2500_02_IOCFG0,   0x01);
-    cc2500WriteReg(CC2500_18_MCSM0,    0x18);
-    cc2500WriteReg(CC2500_07_PKTCTRL1, 0x04);
-    cc2500WriteReg(CC2500_3E_PATABLE,  0xFF);
-    cc2500WriteReg(CC2500_0C_FSCTRL0,  0x00);
-    cc2500WriteReg(CC2500_0D_FREQ2,    0x5C);
-    cc2500WriteReg(CC2500_13_MDMCFG1,  0x23);
-    cc2500WriteReg(CC2500_14_MDMCFG0,  0x7A);
-    cc2500WriteReg(CC2500_19_FOCCFG,   0x16);
-    cc2500WriteReg(CC2500_1A_BSCFG,    0x6C);
-    cc2500WriteReg(CC2500_1B_AGCCTRL2, 0x03);
-    cc2500WriteReg(CC2500_1C_AGCCTRL1, 0x40);
-    cc2500WriteReg(CC2500_1D_AGCCTRL0, 0x91);
-    cc2500WriteReg(CC2500_21_FREND1,   0x56);
-    cc2500WriteReg(CC2500_22_FREND0,   0x10);
-    cc2500WriteReg(CC2500_23_FSCAL3,   0xA9);
-    cc2500WriteReg(CC2500_24_FSCAL2,   0x0A);
-    cc2500WriteReg(CC2500_25_FSCAL1,   0x00);
-    cc2500WriteReg(CC2500_26_FSCAL0,   0x11);
-    cc2500WriteReg(CC2500_29_FSTEST,   0x59);
-    cc2500WriteReg(CC2500_2C_TEST2,    0x88);
-    cc2500WriteReg(CC2500_2D_TEST1,    0x31);
-    cc2500WriteReg(CC2500_2E_TEST0,    0x0B);
-    cc2500WriteReg(CC2500_03_FIFOTHR,  0x07);
-    cc2500WriteReg(CC2500_09_ADDR,     0x00);
+
+    if (RX_SPI_FRSKY_D || RX_SPI_FRSKY_X || RX_SPI_FRSKY_X_LBT) {
```

Comment:
What are you checking here?

Reply:
I was moving the shared register writes for FRSKY_D, FRSKY_X and LBT. Under that conditional check because REDPINE didn't use them.  I could move all the register initialization for redpine to the cc2500_redpine.c if that would be better?

Reply:
But you are or-ing 3 defines without comparing them to anything. :laughing:

Reply:
But yes, it is probably easier to read if only the shared parts are in here and the one-protocol-only stuff goes into the protocol specific files.

Reply:
Oh geesh! yeah that was dumb, I scanned right over that.  I'll move the regisiter init over to redpine.c.  Should I move all the redpine stuff in frsky_shared.c (including binding) over to redpine.c as well?  That would leave frsky_shared.c with no modifications, and would probably look cleaner in the code.

Reply:
I think that would make sense - there does not seem to be that much overlap, and since the FrSky stuff is reverse engineered and based on guesses it might change if there are new findings.